actioncable/lib/action_cable.rb

```diff
@@ -35,7 +35,8 @@ module ActionCable
       confirmation: 'confirm_subscription'.freeze,
       rejection: 'reject_subscription'.freeze
     },
-    default_mount_path: '/cable'.freeze
+    default_mount_path: '/cable'.freeze,
+    protocols: ["actioncable-v1-json".freeze, "actioncable-unsupported".freeze].freeze
```

Comment:
Sorry, that's not right, I know we're not disconnecting server side on protocol mismatch. But can the server set the protocol to `actioncable-unsupported` if the client doesn't open the WebSocket with a supported protocol?

For example, client connects with:
```coffee
@webSocket = new WebSocket(@consumer.url, "actioncable-v1-json")
```
And if the server only knows about `actioncable-v2-json` then it responds by setting the protocol to `actioncable-unsupported`.